tests/Query/DateParser.test.ts

```diff
@@ -144,3 +144,64 @@ describe('Date Parser - correct delta for next & last month & quarter (Today is
         // because the length of Q2 in days is same as Q3.
     });
 });
+
+describe('DateParser - specific date ranges', () => {
+    it('should return dates for specific ranges', () => {
+        // Week (53 in a year)
+        testParsingDateRange('2020-W53', '2020-12-28', '2021-01-03');
+        testParsingDateRange('2021-W01', '2021-01-04', '2021-01-10');
+
+        // Week (52 in a year)
+        testParsingDateRange('2018-W52', '2018-12-24', '2018-12-30');
+        testParsingDateRange('2019-W01', '2018-12-31', '2019-01-06');
+        testParsingDateRange('2019-W02', '2019-01-07', '2019-01-13');
+
+        // Month
+        testParsingDateRange('2002-02', '2002-02-01', '2002-02-28');
+        testParsingDateRange('2013-06', '2013-06-01', '2013-06-30');
+        testParsingDateRange('2017-12', '2017-12-01', '2017-12-31');
+        testParsingDateRange('2024-02', '2024-02-01', '2024-02-29');
+
+        // Quarter
+        testParsingDateRange('2007-Q1', '2007-01-01', '2007-03-31');
+
+        // Year
+        testParsingDateRange('1996', '1996-01-01', '1996-12-31');
+        testParsingDateRange('2022', '2022-01-01', '2022-12-31');
+    });
+
+    it.each([
+        ['2018-W38', '2018-09-17', '2018-09-23'],
+        ['2010-11', '2010-11-01', '2010-11-30'],
+        ['2019-Q3', '2019-07-01', '2019-09-30'],
+        ['2007', '2007-01-01', '2007-12-31'],
+    ])(
+        'specific range %s: should return %s and %s at midnight',
+        (range: string, rangeStart: string, rangeEnd: string) => {
+            const parsedDateRange = DateParser.parseDateRange(range);
+            expect(parsedDateRange[0].format('YYYY-MM-DD HH:mm')).toStrictEqual(`${rangeStart} 00:00`);
+            expect(parsedDateRange[1].format('YYYY-MM-DD HH:mm')).toStrictEqual(`${rangeEnd} 00:00`);
+        },
+    );
+
+    it('should return invalid dates for erroneous specific ranges', () => {
+        // Week (53 & 52 in a year)
+        testParsingDateRange('2020-W54', 'Invalid date', 'Invalid date');
+        testParsingDateRange('2018-W53', 'Invalid date', 'Invalid date');
```

Comment:
I have seen somewhere that some years do have 53 weeks in, depending on where the first day of the year falls... It would be good to understand the reasoning behind this.

So could there be comments saying why these are invalid?

Reply:
I'm confused by the comments referring to 53&52 but the code referring to 54 and 53.

Reply:
Nice. Thanks. Resolving.

Reply:
Oops - wrong one,

Reply:
```
// Week (53 & 52 in a year)
        testParsingDateRange('2020-W54', 'Invalid date', 'Invalid date');
        testParsingDateRange('2018-W53', 'Invalid date', 'Invalid date');
```
Year 2020 has 53 weeks so the 53rd week is valid and 54th week is invalid. For 2018 it is 52, 52 and 53 respectively.

Reply:
I have updated the comment and made it more explicit, pls resolve the conversation if it is clear now